### DISCUSSION TITLE
Does fsnotes use Apple's Advanced Data Protection?

### DISCUSSION THREAD

**Discussion by apastuszak** [glushchenko/fsnotes#1570] [General]
If I have Advanced Data Protection enabled, will all my notes be end-to-end encrypted?

**Comment by gingerbeardman** [comment]
If you store your notes in iCloud, and Apple's feature works as advertised, then yes.

- it's on you to store your notes in iCloud
- it's on Apple for this to work as intended
- FSNotes has no control over either

**Comment by apastuszak** [comment]
According to Apple's document on Advanced Data Protection:

_iCloud stores some data without the protection of user-specific CloudKit service keys, even when Advanced Data Protection is turned on. CloudKit Record fields must be explicitly declared as “encrypted” in the container’s schema to be protected, and reading and writing encrypted fields requires the use of dedicated [APIs](https://developer.apple.com/documentation/cloudkit/ckrecord/3746821-encryptedvalues?language=objc). Dates and times when a file or object was modified are used to sort a user’s information, and checksums of file and photo data are used to help Apple de-duplicate and optimize the user’s iCloud and device storage—all without having access to the files and photos themselves. Details about how encryption is used for specific data categories is available in the Apple Support article [iCloud data security overview](https://support.apple.com/kb/HT202303)._

https://support.apple.com/guide/security/advanced-data-protection-for-icloud-sec973254c5f/web

Does fsnotes use CloudKit or is it just storing notes on iCloud Drive, or both?

**Comment by gingerbeardman** [comment]
Both.

- FSNotes uses CloudKit, [see here](https://github.com/search?q=repo%3Aglushchenko%2Ffsnotes%20cloudkit&type=code)
- FSNotes defaults to using iCloud Drive for storage, but the user can change this

maybe @glushchenko has more thoughts?

**Reply by glushchenko** [reply]
CLOUDKIT was removed many years ago, FSNotes uses iCloud Documents only. And syncs preferences with UserDefaults :)

**Comment by apastuszak** [comment]
That means it is end-to-end encrypted if ADP is turned on.

Why did you stop using CloudKit?

**Reply by gingerbeardman** [reply]
Try a reply rather than a new comment, that way the user you are replying to will get a notification.

**Reply by glushchenko** [reply]
I wanted files to be available in Files.app and synchronized to the cloud, this job is already done by iCloud Documents, there is no need to duplicate this feature in CloudKit.

**Reply by apastuszak** [reply]
Looks like you got the benefits of Advanced Data Protection by going with iCloud Drive instead of CloudKit.
